packages/mobile/src/verify/hooks.ts

```diff
@@ -0,0 +1,29 @@
+import { useAsync } from 'react-async-hook'
+import networkConfig from 'src/geth/networkConfig'
+import Logger from 'src/utils/Logger'
+
+const TAG = 'verify/hooks'
+
+const FETCH_TIMEOUT = 30 * 1000 // 30 secs
+
+const delay = (timeout: number, error: Error) =>
+  new Promise((resolve, reject) => setTimeout(() => reject(error), timeout))
+
+// CB TEMPORARY HOTFIX: helper for Komenci endpoint to ensure availability
+export function useAsyncKomenciReadiness() {
+  return useAsync<boolean>(async () => {
+    Logger.info(TAG, 'Determining komenci readiness...')
+    try {
+      const response = (await Promise.race([
+        fetch(networkConfig.komenciLoadCheckEndpoint),
+        delay(FETCH_TIMEOUT, new Error('Timeout Error')),
+      ])) as Response
+      const isReady = response.ok // status in the range 200-299
+      Logger.info(TAG, `Komenci isReady=${isReady} (statusCode=${response.status})`)
+      return isReady
+    } catch (error) {
+      Logger.error(TAG, 'Failed to determine komenci readiness', error)
+      throw error
```

Comment:
So if the network request fails, this error will propagate and the app will go to the ErrorBoundary fail screen right? Or does useAsync swallow errors? Just confirming the desired behavior.

Reply:
`useAsync` will handle it, i.e. swallow it